src/bp/nlu/engine/engine/index.ts

```diff
@@ -47,7 +46,7 @@ const DEFAULT_ENGINE_OPTIONS: EngineOptions = {
   cacheSize: undefined,
   legacyElection: false
 }
-const DEFAULT_CACHE_SIZE = Infinity
+const MEMORY_SECURITY_FACTOR = 0.4
```

Comment:
not a lot, but just by playing around, I realised the rest of botpress also uses a lot of heap